### PR TITLE
non-json payloads on error responses are handled

### PR DESCRIPTION
**Pull request by f-sander**
https://github.com/dehora/nakadi-java/issues/357

### CODE REVIEW

File: nakadi-java-client/src/main/java/nakadi/Problem.java [hunk 1]
```diff
@@ -145,6 +145,24 @@ public static Problem observerProblem(String title, String detail) {
...
+   * object automatically; typically this happens when a server responds with a non-json payload.
+   *
+   * @param statusCode the HTTP status code of the server's response
+   * @param body the HTTP body of the server's response
+   * @param errorDetail details about why it failed to convert the server's response to a Problem object
+   * @return a Problem object with a status of {@param statusCode} and a type of "about:contract_retryable"
+   */
+  public static Problem rawProblem(int statusCode, String body, String errorDetail) {
+    return new Problem()
+            .title(body)
```
Comment (ePaul) [inline nakadi-java-client/src/main/java/nakadi/Problem.java:159]: We might want to use a substring here in case the body is too long?

**Comment by ePaul** [comment]
In general, this seems like an easy fix for the problem. Thanks for digging into it.

File: nakadi-java-client/src/main/java/nakadi/Problem.java (hunk 1 above)
Comment (f-sander) [reply nakadi-java-client/src/main/java/nakadi/Problem.java:159]: sure, what do you think would be a reasonable safe-guard?

Comment (ePaul) [reply nakadi-java-client/src/main/java/nakadi/Problem.java:159]: Maybe 200 characters?
(At least the "title" name lets me think of something short-ish.)

Comment (f-sander) [reply nakadi-java-client/src/main/java/nakadi/Problem.java]: done

**Comment by adyach** [comment]
@dehora can we have that merged?

**Pull request merged**